lib/kbuildbarn/buddy.go

```diff
@@ -0,0 +1,44 @@
+package kbuildbarn
+
+import (
+	"context"
+	"github.com/enfabrica/enkit/lib/bes"
+	bespb "github.com/enfabrica/enkit/third_party/bazel/buildeventstream"
+)
+
+type FilterOption func(event *bespb.BuildEvent, baseName, invocation, clusterName string) BBClientdList
+
+func WithTestResults() FilterOption {
+	return func(event *bespb.BuildEvent, baseName, invocation, clusterName string) BBClientdList {
+		testResult := event.GetTestResult()
+		if testResult != nil {
+			return GenerateLinksForFiles(testResult.TestActionOutput, baseName, invocation, clusterName)
+		}
+		return nil
+	}
+}
+
+func WithNamedSetOfFiles() FilterOption {
+	return func(event *bespb.BuildEvent, baseName, invocation, clusterName string) BBClientdList {
+		nsof := event.GetNamedSetOfFiles()
+		if nsof != nil {
+			return GenerateLinksForFiles(nsof.Files, baseName, invocation, clusterName)
+		}
+		return nil
+	}
+}
+
+func GenerateSymlinks(client *bes.BuildBuddyClient, baseName, invocation, clusterName string, options ...FilterOption) (BBClientdList, error) {
+	ctx := context.Background()
```

Comment:
This context should be a param - only tests/main should grab the background context

Reply:
Done